CookingBook/src/DataBase/UserDataBase/usersdatabase.cpp

```diff
@@ -0,0 +1,326 @@
+#include "usersdatabase.h"
+
+UsersDataBase::CUsersDataBase* UsersDataBase::CUsersDataBase::m_pUsersDataBase=nullptr;
+int UsersDataBase::CUsersDataBase::m_nAmountOfUsers = 0;
+int UsersDataBase::CUsersDataBase::m_nAmountOfObjects = 0;
+UsersDataBase::CUsersDataBaseDestroyer UsersDataBase::CUsersDataBase::m_Destroyer;
+const char* UsersDataBase::CUsersDataBase::m_chPath = "";
+//UsersDataBase::SUser UsersDataBase::CUsersDataBase::m_user;
+
+//-----------------------------------------------------------------------------
+UsersDataBase::CUsersDataBase::CUsersDataBase()
+{
+	m_chPath = nullptr;
+}
+//-----------------------------------------------------------------------------
+UsersDataBase::CUsersDataBase::CUsersDataBase(const char* chPath)
+{
+	m_chPath = const_cast<char*>(chPath);
+}
+//-----------------------------------------------------------------------------
+UsersDataBase::CUsersDataBase::~CUsersDataBase()
+{
+}
+//-----------------------------------------------------------------------------
+void UsersDataBase::CUsersDataBase::SetAmountOfUsers()
+{
+	sqlite3_stmt* InfoFromDB;
+	sqlite3* sqliteDataBase;
+	int nStatus = sqlite3_open(m_chPath, &sqliteDataBase);
+	if (nStatus != SQLITE_OK) {
+		std::cerr << "Error opening table from SetAmountOfUsers" << std::endl;
+	}
+	else {
+		std::string sQuery = "SELECT COUNT(*) FROM USERS";
+		nStatus = sqlite3_prepare_v2(sqliteDataBase, sQuery.c_str(), READ_UP_TO_ZERO_TERMINATOR, &InfoFromDB, NULL);
+		if (nStatus != SQLITE_OK) {
+			std::cout << "error " << sqlite3_errmsg(sqliteDataBase);
+		}
+		else {
+			while ((nStatus = sqlite3_step(InfoFromDB) == SQLITE_ROW)) {
+				m_nAmountOfUsers = sqlite3_column_int(InfoFromDB, 0);
+			}
+		}
+		sqlite3_finalize(InfoFromDB);
+	}
+}
+//-----------------------------------------------------------------------------
+int UsersDataBase::CUsersDataBase::GetAmountOfUsers()
+{
+	return m_nAmountOfUsers;
+}
+//-----------------------------------------------------------------------------
+bool UsersDataBase::CUsersDataBase::CreateDB()
+{
+	sqlite3* sqliteDataBase;
+	int nExit = sqlite3_open(m_chPath, &sqliteDataBase);
+
+	if (nExit == SQLITE_OK) {
+		std::cout << "sqliteDataBase created successfully";
+	}
+	else {
+		std::cout << "Error creation sqliteDataBase";
+	}
+	sqlite3_close(sqliteDataBase);
+	return true;
```

Comment:
If your function return only true value. There is no reason to make it bool. You can make it void.
But seems like  it can return false in else block